### ISSUE TITLE
Inconsistent starting position of chain method continuation

### ISSUE THREAD

**Issue by hanggrian** [pinterest/ktlint#2737]
When a method owner is constant or literal, the rule expects the first call to start on the next line.

```kt
""
    .plus(1)
    .plus(2)
    .plus(3)
    .plus(4)
```

Using the same convention, the rule reports an error when the receiver is conditional `if` or `when`.

```kt
when {
    true -> ""
    else -> ""
} // VIOLATION
    .plus(1)
    .plus(2)
    .plus(3)
    .plus(4)

if (true) {
    ""
} else {
    ""
} // VIOLATION
    .plus(1)
    .plus(2)
    .plus(3)
    .plus(4)
```

## Expected Behavior

The first method call should always start on the next line. Alternatively, the rule could allow the first call in the same line. [Kotlin coding conventions](https://kotlinlang.org/docs/coding-conventions.html#wrap-chained-calls) state that it's OK to omit the initial line break "if it makes more sense that way".

```kt
""
    .plus(1)
    .plus(2)
    .plus(3)
    .plus(4)

when {
    true -> ""
    else -> ""
}
    .plus(1)
    .plus(2)
    .plus(3)
    .plus(4)

// or

"".plus(1)
    .plus(2)
    .plus(3)
    .plus(4)

when {
    true -> ""
    else -> ""
}.plus(1)
    .plus(2)
    .plus(3)
    .plus(4)
```

## Current Behavior

```kt
""
    .plus(1)
    .plus(2)
    .plus(3)
    .plus(4)

when {
    true -> ""
    else -> ""
}.plus(1)
    .plus(2)
    .plus(3)
    .plus(4)
```

## Additional information

- Current version of ktlint: 1.3.1

**Comment by paul-dingemans** [comment]
In example below, the `.plus(1)` seems to be disconnected from the root of the method continuation:
```
when {
    true -> ""
    else -> ""
}
    .plus(1)
    .plus(1)
```

while in code below the chain operators are not aligned:
```
when {
    true -> ""
    else -> ""
}.plus(1)
    .plus(1)
```

> [Kotlin coding conventions](https://kotlinlang.org/docs/coding-conventions.html#wrap-chained-calls) state that it's OK to omit the initial line break "if it makes more sense that way".

Statements like above, can result in discussions on a case by case base how the statement would be written. Both styles could be in use within the same code base, file, class or method. Ktlint tries to avoid inconsistencies, and has (fairly arbitrarly) choosen the second style to be applied always.

Let me be clear, I dislike both ways of writing code like above equally. Personally, I would tend to refactor my code by extracting the when-clause to solve the formatting issue. It is not pretty when ktlint (or whatever formatter) more or less forces you to refactor code, but on the other hand to might see it as a removing a code smell that makes the code easier to read.
